client/db_test.go

```diff
@@ -0,0 +1,75 @@
+package client_test
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/cockroachdb/cockroach/client"
+	"github.com/cockroachdb/cockroach/server"
+)
+
+func ExampleDB() {
+	s := server.StartTestServer(nil)
+	defer s.Stop()
+
+	db := client.Open("https://root@" + s.ServingAddr())
+
+	if err := db.Put("key", "value"); err != nil {
+		log.Fatal(err)
+	}
+	result := db.Get("key")
+	fmt.Println(1, result.Rows[0].ValueStringer())
+
+	if r := db.Inc("inc", 100); r.Err != nil {
+		log.Fatal(r.Err)
+	}
+	result = db.Get("key", "inc")
+	fmt.Println(2, result.Rows[0].ValueStringer())
+	fmt.Println(3, result.Rows[1].ValueInt())
+
+	b := &client.Batch{}
+	b.Get("key")
+	b.Scan("i", "j", 100)
+	b.Put("foo", "bar")
+	if err := db.Run(b); err != nil {
+		log.Fatal(err)
+	}
+	fmt.Println(4, b.Results[0].Rows[0].ValueStringer())
```

Comment:
cool. This looks pretty smooth already. This part would probably bloat in actual code:

```
if len(b.Rows) > 0 { ... }
```

Maybe something like

`val, ok := b.Results[0].ValueStringer()`?
`i, ok := b.Results[1].ValueInt()`?

to get the first result (if there is one), and for other indexes
`val_i, ok := b.ValueInt(i)`

and for slices

`val_two_and_three_as_slice, ok := b.Results.ValueBytes(2, 3)`

Just brainstorming here.